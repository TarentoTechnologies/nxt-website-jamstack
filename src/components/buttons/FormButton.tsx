import { bgHaysBlue } from "../../styles/style-guide/ColorGuide.module.css";
import { buttonClass } from "./FormButton.module.css";

interface FormButtonProps {
  label: any;
  primary?: boolean;
  clickHandler?: (event: any) => void;
  btnLink?: string;
  isEmail: boolean;
}

export const FormButton = ({
  label,
  primary = false,
  clickHandler,
  btnLink,
  isEmail,
}: FormButtonProps) => {
  return (
    <a href={btnLink != null ? (isEmail ? `mailto:${btnLink}` : btnLink) : ""}>
      <div
        className={`text-center d-flex align-items-center justify-content-center 
      ${buttonClass} ${bgHaysBlue}`}
        onClick={clickHandler}
      >
        <label>{label}</label>
      </div>
    </a>
  );
};
