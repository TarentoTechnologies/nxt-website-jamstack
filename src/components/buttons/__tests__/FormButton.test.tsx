import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { FormButton } from "../FormButton";

describe("Test if the FormButton component is working as expected", () => {
  it("<FormButton /> matches snapshot", () => {
    const component = render(
      <FormButton data-testid="form-button" label="Request a Demo" isEmail />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<FormButton /> renders without exploding", () => {
    const component = render(
      <FormButton data-testid="form-button" label="Request a Demo" isEmail />
    );
    expect(component.getByTestId("form-button")).toBeInTheDocument();
  });
});
