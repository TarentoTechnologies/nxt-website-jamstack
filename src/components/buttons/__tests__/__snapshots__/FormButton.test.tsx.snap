// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the FormButton component is working as expected <FormButton /> matches snapshot 1`] = `
<div>
  <a
    data-testid="form-button"
    href=""
  >
    <div
      class="text-center d-flex align-items-center justify-content-center 
      buttonClass bgHaysBlue "
      data-aos="fade-up"
    >
      <label
        data-aos="fade-up"
      >
        Request a Demo
      </label>
    </div>
  </a>
</div>
`;
